bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/handler/JuiceNetDeviceHandler.java

```diff
@@ -0,0 +1,295 @@
+                    logger.debug("DateTime: {} - {}", datetime.toString(), targetTime);
+
+                    getApi().setOverride(token, energyAtPlugin, targetTime, energyToAdd);
+
+                    break;
+                }
+                case DEVICE_CHARGING_STATE: {
+                    String state = ((StringType) command).toString();
+                    Long overrideTime = deviceStatus.unit_time;
+                    int energyAtPlugin = 0;
+                    int energyToAdd = deviceCar.battery_size_wh;
+
+                    switch (state) {
+                        case "stop":
+                            if (targetTimeTou == 0) {
+                                targetTimeTou = deviceStatus.target_time;
+                            }
+                            overrideTime = deviceStatus.unit_time + 31556926;
+                            break;
+                        case "start":
+                            if (targetTimeTou == 0) {
+                                targetTimeTou = deviceStatus.target_time;
+                            }
+                            overrideTime = deviceStatus.unit_time;
+                            break;
+                        case "smart":
+                            overrideTime = deviceStatus.default_target_time;
+                            break;
+                    }
+
+                    getApi().setOverride(token, energyAtPlugin, overrideTime, energyToAdd);
+
+                    break;
+                }
+            }
+        } catch (IOException e) {
+            logger.debug("Unable to open connection to api host: {}", e.getMessage());
+        } catch (InterruptedException e) {
+            logger.debug("Unable to open connection to api host: {}", e.getMessage());
+        } catch (JuiceNetApiException e) {
+            logger.debug("Malformed JuiceNet API error: {}", e.getMessage());
+        }
+    }
+
+    public void queryDeviceStatusAndInfo() throws JuiceNetApiException, IOException, InterruptedException {
+        deviceStatus = getApi().queryDeviceStatus(token);
+
+        if (deviceStatus.info_timestamp > lastInfoTimestamp) {
+            lastInfoTimestamp = deviceStatus.info_timestamp;
+            deviceInfo = getApi().queryInfo(token);
+            deviceTouSchedule = getApi().queryTOUSchedule(token);
+            refreshInfoChannels();
+        }
+
+        int carId = deviceStatus.car_id;
+        for (JuiceNetApi.JuiceNetApiCar car : deviceInfo.cars) {
+            if (car.car_id == carId) {
+                this.deviceCar = car;
+                break;
+            }
+        }
+
+        refreshStatusChannels();
+    }
+
+    protected ZonedDateTime toZonedDateTime(long localEpochSeconds) {
+        Instant instant = Instant.ofEpochSecond(localEpochSeconds);
+        ZonedDateTime zdt = instant.atZone(ZoneId.of("UTC"));
+
+        return zdt.withZoneSameLocal(ZoneId.systemDefault());
```

Comment:
If this ZonedDateTime is exposed to channels, please use `TimeZoneProvider` for getting configured time zone.